File: rzops-api/domain/src/models/attachment.rs
use chrono::{DateTime, Utc};
use uuid::Uuid;

use crate::enums::ReservedStatus;

/// Attachment entity — maps to cmdb_attachment.
#[derive(Debug, Clone)]
pub struct Attachment {
    pub id: Uuid,
    pub filename: String,
    pub target_type: Option<String>,
    pub target_id: Option<Uuid>,
    pub storage_key: Option<String>,
    pub content_type: Option<String>,
    pub size_bytes: Option<i64>,
    pub uploaded_by_id: Option<Uuid>,
    pub status: ReservedStatus,
    pub remarks: Option<String>,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}

File: rzops-web/src/lib/types/attachment.ts
export interface AttachmentResponse {
  id: string;
  filename: string;
  target_type: string | null;
  target_id: string | null;
  target_name: string | null;
  storage_key: string | null;
  content_type: string | null;
  size_bytes: number | null;
  uploaded_by_id: string | null;
  uploader_name: string | null;
  status: string;
  remarks: string | null;
  created_at: string;
  updated_at: string;
}

export interface AttachmentListResponse {
  data: AttachmentResponse[];
  count: number;
}

export interface CreateAttachmentRequest {
  filename: string;
  target_type?: string;
  target_id?: string;
  storage_key?: string;
  content_type?: string;
  size_bytes?: number;
  uploaded_by_id?: string;
  status?: string;
  remarks?: string;
}

export interface ListAttachmentsQuery {
  q?: string;
  status?: string;
  target_type?: string;
  target_id?: string;
  page?: number;
  per_page?: number;
}
